### PR TITLE
PMM-5895 Basic kubernetes client.

### PR DESCRIPTION
**Pull request by BupycHuk**

### CODE REVIEW

File: service/k8sclient/k8client.go [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+
+		cluster := Cluster{
+			Status: "deleting",
+			Kind:   kind,
+			Name:   clusterName,
+		}
+		res = append(res, cluster)
+
+		exists[clusterName] = struct{}{}
+	}
```
[the remark this answers is not in the document]
Comment (BupycHuk) [reply service/k8sclient/k8client.go:278]: I ran gofumpt locally, but it's still here.

File: .golangci.yml [hunk 2]
```diff
@@ -27,6 +27,7 @@ linters:
     - gomnd     # too annoying
     - lll       # gRPC types are long
     - wsl       # too annoying
+    - unused    # too many false positives
```
Comment (AlekSi) [inline .golangci.yml:30]: It actually doesn't have false positives: we don't use that code outside of the tests, and linter is doing a good job checking that we don't keep unused code.

Let's keep it there, but I will revert that change after merging.

File: service/k8sclient/k8client.go [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+	Name string
+	Size int32
+}
+
+// Cluster contains information related to cluster.
+type Cluster struct {
+	Name   string
+	Kind   ClusterKind
+	Size   int32
+	Status string // FIXME
```
[the remark this answers is not in the document]
Comment (AlekSi) [reply service/k8sclient/k8client.go:58]: This should be an own type too

File: service/k8sclient/k8client.go [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+	}
+}
+
+// Cleanup removes temporary files created by that object.
+func (c *K8Client) Cleanup() {
+	c.kubeCtl.Cleanup()
+}
+
+// ListClusters returns list of clusters and their statuses.
+func (c *K8Client) ListClusters(ctx context.Context) ([]Cluster, error) {
```
Comment (AlekSi) [inline service/k8sclient/k8client.go:79]: There should two method for two types of clusters – they have different properties like ProxySQL settings for PXC

File: service/k8sclient/k8client.go [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+	res := append(perconaXtraDBClusters, deletingClusters...)
+
+	return res, nil
+}
+
+// CreateXtraDBCluster creates Percona XtraDB cluster with provided parameters.
+func (c *K8Client) CreateXtraDBCluster(ctx context.Context, params *XtraDBParams) error {
+	res := &pxc.PerconaXtraDBCluster{
+		TypeMeta: meta.TypeMeta{
+			APIVersion: "pxc.percona.com/v1-4-0",
```
Comment (AlekSi) [inline service/k8sclient/k8client.go:100]: That should be a named constant

File: service/k8sclient/kubectl/kubectl.go [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+// It returns `errNotFound` if object is not found.
+func (k *KubeCtl) Get(ctx context.Context, kind string, name string, res interface{}) error {
+	args := []string{"get", "-o=json", kind}
+	if name != "" {
+		args = append(args, name)
+	}
+
+	stdout, stderr, err := k.run(ctx, args, nil)
+	if err != nil {
+		if strings.Contains(string(stderr), "not found") { // FIXME
```
Comment (AlekSi) [inline service/k8sclient/kubectl/kubectl.go:80]: Can we do better?

File: service/k8sclient/kubectl/kubectl.go [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+	var outBuf bytes.Buffer
+	var errBuf bytes.Buffer
+	cmd := exec.CommandContext(ctx, args[0], args[1:]...) //nolint:gosec
+	pdeathsig.Set(cmd, unix.SIGKILL)
+	cmd.Stdin = &inBuf
+	cmd.Stdout = &outBuf
+	cmd.Stderr = &errBuf
+	err := cmd.Run()
+
+	k.l.Debugf(outBuf.String()) // FIXME
```
Comment (AlekSi) [inline service/k8sclient/kubectl/kubectl.go:127]: That will not work if output contains `%s` or something similar. We can't use XXXf functions as just XXX.

File: service/k8sclient/kubectl/kubectl.go (hunk 6 above)
Comment (BupycHuk) [reply service/k8sclient/kubectl/kubectl.go]: Removed it since we don't need it anymore.

File: service/k8sclient/kubectl/kubectl.go (hunk 7 above)
Comment (BupycHuk) [reply service/k8sclient/kubectl/kubectl.go:135]: Added new Debug method.

Comment (AlekSi) [reply service/k8sclient/kubectl/kubectl.go]: Why add more methods to interface and implementations instead of using `Debugf("%s", outBuf)`?

Comment (BupycHuk) [reply service/k8sclient/kubectl/kubectl.go]: ok, anything else?

Comment (BupycHuk) [reply service/k8sclient/kubectl/kubectl.go]: Added other methods, since we already have the same usages for Fatal, Info and Debug.

**Pull request merged**